test/e2e/snaps/test-snap-cronjob.spec.js

```diff
@@ -33,34 +28,30 @@ describe('Test Snap Cronjob', function () {
         // added delay for firefox (deflake)
         await driver.delayFirefox(1000);
 
-        // wait for and click connect
-        await driver.waitForSelector('#connectcronjobs');
+        // click connect
         await driver.clickElement('#connectcronjobs');
 
         // switch to metamask extension
         await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-        // wait for and click connect
-        await driver.waitForSelector({
-          text: 'Connect',
-          tag: 'button',
-        });
+        // click connect
         await driver.clickElement({
           text: 'Connect',
           tag: 'button',
         });
 
+        await driver.waitForSelector({
+          tag: 'h3',
+          text: 'Add to MetaMask',
+        });
```

Comment:
this has been fixed in this PR and not in[ this other one](https://github.com/MetaMask/metamask-extension/pull/33012), because changing this spec in the other PR, would have triggered x5 re-runs, and given that there is this other source of flakiness that would likely fail several times. So this fix is added here together with the window fix